src/commands/ui.rs

```diff
@@ -11,36 +17,63 @@ use crate::commands::ExitCode;
 use crate::options::{Options, UI};
 use crate::platform::data_dir;
 use crate::portable::local::{instance_data_dir, InstanceInfo};
+use crate::portable::ver::Specific;
 use crate::print;
 
 pub fn show_ui(options: &Options, args: &UI) -> anyhow::Result<()> {
     let connector = options.create_connector()?;
     let builder = connector.get()?;
     let mut url = format!("http://{}:{}/ui", builder.get_host(), builder.get_port());
-    if let Some(instance) = builder.get_instance_name() {
-        if instance != "_localdev" {
-            let ver = InstanceInfo::read(instance)?.get_version()?.specific();
-            if ver.major < 2 {
-                print::error(format!(
-                    "the specified instance runs EdgeDB v{}, which does not \
-                    include the Web UI; consider upgrading the instance to \
-                    EdgeDB 2.x or later",
-                    ver,
-                ));
-                return Err(ExitCode::new(2).into())
-            }
+
+    if !args.skip_version_check {
+        let version = if let Some(meta) = builder
+            .get_instance_name()
+            .map(InstanceInfo::try_read)
+            .unwrap_or(Ok(None))?
+        {
+            meta.get_version()?.specific()
+        } else {
+            task::block_on(get_remote_version(builder))?
+        };
+        if version.major < 2 {
+            print::error(format!(
+                "the specified instance runs EdgeDB v{}, which does not \
+                include the Web UI; consider upgrading the instance to \
+                EdgeDB 2.x or later",
+                version,
+            ));
+            return Err(ExitCode::new(2).into());
         }
-        match generate_jwt(instance) {
-            Ok(token) => {
-                url = format!("{}?authToken={}", url, token);
-            }
+    }
+
+    if let Some(instance) = builder.get_instance_name() {
+        match if cfg!(windows) {
+            crate::portable::windows::read_jose_keys(instance)
+        } else {
+            read_jose_keys(instance)
+        } {
+            Ok(keys) => match generate_jwt(keys) {
+                Ok(token) => {
+                    url = format!("{}?authToken={}", url, token);
+                }
+                Err(e) => {
+                    print::warn(format!("Cannot generate authToken: {:#}", e));
+                }
+            },
             Err(e) => {
-                print::warn(format!("Cannot generate authToken: {:#}", e));
+                // Ignore the error assuming it's a remote instance
```

Comment:
Please emit a special error and use `e.is::<NonLocalInstance>()`. Swallowing all the errors will make it hard to debug some non-obvious case.